Files/homework/DiskAnalyzer/src/main/java/com/epam/files/FileFinder.java

```diff
@@ -0,0 +1,110 @@
+package com.epam.files;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class FileFinder {
+    private FileFinder() {};
+
+    public static Optional<Path> findFileWithMaxS(Path path) {
+        long maxCount = 0;
+        Path maxFile = null;
+
+        // Traverse paths and find one with max number of 's'
+        try (Stream<Path> stream = Files.walk(path)) {
+            for (Path file : stream.toList()) {
+                if (!Files.isDirectory(file)) {
+                    long count = countLetterS(file);
+                    if (count > maxCount) {
+                        maxCount = count;
+                        maxFile = file;
+                    }
+                }
+            }
+            return Optional.ofNullable(maxFile);
+        } catch (Exception e) {
+            return Optional.empty();
+        }
+    }
+
+    private static long countLetterS(Path file) {
+        return countLetterS(file.getFileName().toString());
+    }
+
+    private static long countLetterS(String filename) {
+        return filename.toLowerCase().chars()
+                .filter(c -> c == 's')
+                .count();
+    }
+
+    public static List<Path> findTop5LargestFiles(Path path) {
+        // Traverse to find top 5 largest files
+        try (Stream<Path> stream = Files.walk(path)) {
+            return stream
+                    // Exclude count directories
+                    .filter(p -> !Files.isDirectory(p))
+                    // Sort by size
+                    .sorted((p1, p2) -> {
```

Comment:
Just a few thoughts:
The behavior of the sorted + limit is actually not deterministic in the specification. In some cases, it may be necessary to use iteration with some of a k-nth min/max algorithms (https://www.geeksforgeeks.org/kth-smallest-largest-element-in-unsorted-array/)